### PR TITLE
add scrolling to widget

### LINKED ISSUE #3: Add scrolling when track title too long
The widget needs to be able to scroll the full song title when it's too long to fit in the player widget, to be compatible with the old widget.

### PR DESCRIPTION
**Pull request by aidenwallis**
### Description

Allows the widget to scroll when the length of the text is longer than the parent element.

Fixes #3

### CODE REVIEW

**Review by mmattbtw — APPROVED** [review]
MTGL

File: src/client/widget/index.tsx [hunk 1]
```diff
@@ -54,37 +54,58 @@ const TitleVisibilityWrapper = ({ children, visibility }: PropsWithChildren<{ vi
   return <div className={`transition-all ease-in-out duration-200 ${classes[visibility]}`}>{children}</div>;
 };
 
+const MAX_TITLE_WIDTH = 236;
+
+const TitleScroller = ({ children }: PropsWithChildren<{}>) => {
```
Comment (WalshyDev) [inline src/client/widget/index.tsx:59]: Missing unit tests

**Review by WalshyDev — CHANGES_REQUESTED** [review]

**Comment by treuks** [comment]
🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀

File: src/client/widget/index.tsx [hunk 2]
```diff
@@ -54,37 +54,58 @@ const TitleVisibilityWrapper = ({ children, visibility }: PropsWithChildren<{ vi
...
+  const [scrolling, setScrolling] = useState(false);
+  const ref = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    ref?.current && setScrolling(ref.current.offsetWidth > MAX_TITLE_WIDTH);
+  }, [ref?.current?.offsetWidth, setScrolling]);
+
+  return (
+    <div ref={ref} className={`absolute${scrolling ? " " + styles.scroll : ""}`}>
```
Comment (WalshyDev) [inline src/client/widget/index.tsx:68]: doesn't use marquee

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/client/widget/index.tsx (hunk 1 above)
Comment (gempir) [inline src/client/widget/index.tsx:59]: PropsWithChildren is not recommended to be used anymore. Coming directly from Dan Abramov

**Review by gempir — CHANGES_REQUESTED** [review]

File: src/client/widget/index.tsx (hunk 1 above)
Comment (aidenwallis) [reply src/client/widget/index.tsx:59]: lol

**Pull request merged**